fix: avoid glob expansion errors when no node versions installed

tennisleng opened the pull request:
## Summary
Fixes #3727

When no node versions are installed, `nvm ls` produces errors like:
```
find: /path/.nvm/versions/node/*: No such file or directory
```

This happens because bash expands `*` literally when `nullglob` is not set and the directory is empty.

## Solution
- Build find arguments dynamically without shell glob expansion
- Use `find DIR -mindepth 1 -maxdepth 1` instead of `find DIR/*`
- Suppress stderr from find to handle edge cases gracefully
- Deduplicate directories before passing to find

## Before
```bash
$ nvm ls
find: /Users/balupton/.nvm/versions/node/*: No such file or directory
find: /Users/balupton/.nvm/versions/node/*: No such file or directory
-> system
```

## After
```bash
$ nvm ls
-> system
```

## Testing
- [x] Verified with empty `~/.nvm/versions/node/` directory
- [x] Verified with existing node versions installed

Commit:
fix: enable nullglob to avoid glob expansion errors when no node versions installed

Fixes #3727

When no node versions are installed, nvm ls produces errors like:
`find: /path/.nvm/versions/node/*: No such file or directory`

This happens because bash expands `*` literally when nullglob is not
set and the directory is empty.

Solution:
- For bash: temporarily enable nullglob before the find command
- For other shells (sh, dash, zsh): suppress stderr (2>/dev/null)
- Restore nullglob to its previous state after the command completes
- Uses BASH_VERSION check to detect bash vs other shells

ljharb commented:
Instead of altering a bunch of the logic, could we perhaps just disable nullglob when needed for the `nvm_ls` invocation?

ljharb commented:
so this kind of works, but relies on nothing exiting prior to the restoration.

i was thinking instead, of wrapping every call to `nvm_ls` in this nullglob check, and turning it off in a *subshell* in which nvm_ls is invoked - that way, it's impossible have the nullglob change persist.

Commit:
refactor: use subshell to isolate nullglob changes

Move nullglob enabling into the command substitution subshell so that
the shell option change cannot persist to the parent shell, even if
something exits early. This is a safer approach than save/restore.

The subshell automatically cleans up when it exits, making it impossible
for nullglob to leak into the user's shell environment.

Commit:
test: add regression test for nullglob issue #3727

Adds a test that explicitly disables nullglob and runs nvm ls with
an empty versions directory to ensure the fix prevents regressions.

The test verifies:
- nvm ls does not produce 'No such file or directory' errors
- nvm ls exits with code 0 when no versions are installed

ljharb reviewed (changes requested):
Unfortunately the test passes even without the fix.
